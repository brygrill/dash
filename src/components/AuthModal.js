// @flow
import React, { Component } from 'react';
import Dialog from 'material-ui/Dialog';
import Button from 'material-ui/Button';

export default class AuthModal extends Component {
  static defaultProps = {
    disabled: false,
    modal: false,
    open: false,
  };

  props: {
    open?: false,
    title: string,
    sub: string,
    disabled?: boolean,
    modal?: boolean,
    submitBtnLabel: string,
    handleCancel: Function,
    handleSubmit: Function,
    alternative: any,
  };

  render() {
    const {
      open,
      title,
      sub,
      disabled,
      modal,
      submitBtnLabel,
      handleCancel,
      handleSubmit,
      alternative,
    } = this.props;

    return (
      <div>
        <Dialog open={open} modal={modal} onRequestClose={handleCancel}>
          <div className="sbr-align-center ">
            <h3 className="sbr-margin-bottom-0">{title} </h3>
            <h6 className="sbr-margin-top-half">{sub}</h6>
            <Button primary disabled={disabled} onTouchTap={handleSubmit}>
              {submitBtnLabel}
            </Button>
            <div>{alternative}</div>
          </div>
        </Dialog>
      </div>
    );
  }
}
