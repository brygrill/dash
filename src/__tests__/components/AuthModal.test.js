import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import AuthModal from '../../components/AuthModal';

describe('<AuthModal />', () => {
  it('Renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <MuiThemeProvider><AuthModal open /></MuiThemeProvider>,
      div,
    );
  });
  it('Matches snapshot with child', () => {
    const component = renderer.create(
      <MuiThemeProvider><AuthModal open /></MuiThemeProvider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
