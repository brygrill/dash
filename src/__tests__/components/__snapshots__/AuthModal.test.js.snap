// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthModal /> Matches snapshot with child 1`] = `<div />`;
